server.go

```diff
@@ -1541,6 +1555,73 @@ func (c *createMeasurementsIfNotExistsCommand) addFieldIfNotExists(measurement,
 	return nil
 }
 
+func (s *Server) applyDropSeries(m *messaging.Message) error {
+	var c dropSeriesCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	database := s.databases[c.Database]
+	if database == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// Remove from metastore.
+	for _, seriesID := range c.SeriesIDs {
+		series := database.series[seriesID]
+		if series == nil {
+			return fmt.Errorf("series not found for id %d", seriesID)
+		}
+		m := series.measurement
+		if m == nil {
+			return fmt.Errorf("measurement not found for series %d", seriesID)
+		}
+		err := s.meta.mustUpdate(func(tx *metatx) error { return tx.dropSeries(c.Database, m.Name, seriesID) })
+		if err != nil {
+			return err
+		}
+	}
+
+	// Remove shard data
+	for _, rp := range s.databases[c.Database].policies {
```

Comment:
If you actually think this stuff could fail, you should wrap it in the transaction above, which will mean it will all work, or the server will restart. Otherwise one could end up with a Metastore with one thing, but in-RAM info saying something else. I would not like to see a server continuing in this state. Check with @benbjohnson if you would like to learn more, since he and I discussed a similar issue with regards to batching.